src/os_image_nodelet.cpp

```diff
@@ -52,7 +52,14 @@ class OusterImage : public nodelet::Nodelet {
     void create_lidar_packets_subscriber() {
         lidar_packet_sub = getNodeHandle().subscribe<PacketMsg>(
             "lidar_packets", 100, [this](const PacketMsg::ConstPtr msg) {
-                if (lidar_packet_handler) lidar_packet_handler(msg->buf.data());
+                if (lidar_packet_handler) {
+                    // TODO[UN]: this is not ideal since we can't reuse the msg buffer
+                    // Need to redefine the Packet object and allow use of array_views
+                    sensor::LidarPacket lidar_packet(msg->buf.size());
```

Comment:
I think you can actually get ROS to decode a message into a custom data type to avoid this copy (probably).

Reply:
https://wiki.ros.org/roscpp/Overview/MessagesSerializationAndAdaptingTypes

Reply:
I am not familiar with this option, I need to look into it to see if it can help with the situation here